guacamole/src/main/webapp/app/navigation/directives/guacUserMenu.js

```diff
@@ -134,6 +134,48 @@ angular.module('navigation').directive('guacUserMenu', [function guacUserMenu()
                 return authenticationService.isAnonymous();
             };
 
+            /**
+             * Requests fullscreen for the Guacamole page.
+             */
+            $scope.fullscreen = function fullscreen() {
+                var elem = document.documentElement;
+                var fullscreenElem = !(!document.fullscreenElement &&
+                    !document.msFullscreenElement &&
+                    !document.mozFullScreenElement &&
+                    !document.webkitFullscreenElement);
+
+                if (!fullscreenElem) {
+                    if (!elem.requestFullscreen) {
+                        elem.requestFullscreen = (elem.mozRequestFullScreen
+                                || elem.webkitRequestFullscreen
+                                || elem.msRequestFullscreen).bind(elem);
+                    }
```

Comment:
It would be better to retrieving `document.documentElement` and ensure its `requestFullScreen()` and `exitFullscreen()` are defined just once, rather than each time `$scope.fullscreen()` is invoked. If these checks were moved into the body of the controller rather than the body of this function, I think this would become more readable (and would happen to also do less work).

Reply:
@mike-jumper If we move the tests for `requestFullScreen()`  and `exitFullscreen()` into the body, then we could detect if the current browser has support for fullscreen before adding the function to the menu.

Should we still display the fullscreen menu or is it better to hide it ?

Reply:
Definitely better to hide it.